post-scan-actions/aws-python-teams-notification/handler.py

```diff
@@ -0,0 +1,50 @@
+import urllib3
+import json
+import os
+http = urllib3.PoolManager()
+import textwrap
+
+def lambda_handler(event, context):
+
+    url = os.environ['TEAMS_URL']
+    channel = os.environ['TEAMS_CHANNEL']
+    
+    for record in event['Records']:
+
+        #Message details from SNS event
+        message = json.loads(record['Sns']['Message'])
+        findings = message['scanning_result'].get('Findings')
+
+        #ARN info to get AWS Account ID
+        arn = json.dumps(record['EventSubscriptionArn'])
+        account_id = arn.split(":")[4].strip()
+
+        if findings:
+        
+            malwares = []
+            types = []
+            for finding in message['scanning_result']['Findings']:
+                malwares.append(finding.get('malware'))
+                types.append(finding.get('type'))
+
+            body_text = textwrap.dedent('''
+            WARNING
+            AWS Account ID: {account}
+            File URL: {file_url}
+            Malware Name(s): {malwares}
+            Malware Type(s): {types}
+            ''').format(
+            account=str(account_id),
+            file_url=str(message['file_url']),
+            malwares=', '.join(malwares),
+            types=', '.join(types)
+            )
+            
+            payload = {
+                        "channel": channel,
+                        "text": body_text,
+                        "icon_emoji": ":rotating_light:"
+                        }
```

Comment:
According to [Teams' document](https://docs.microsoft.com/en-us/outlook/actionable-messages/adaptive-card), webhook payload format does not have `channel` and `icon_emoji`. Please follow the standard format to compose the payload. F.Y.R., [Message card playground](https://messagecardplayground.azurewebsites.net/) contains some examples. B.T.W, using `facts` to list information looks better than plain text in the message cards.

Reply:
I have updated payload for using the card format for ms teams